parquet/src/main/java/org/apache/iceberg/parquet/vectorized/VectorizedParquetReader.java

```diff
@@ -0,0 +1,290 @@
+      this.model = readerFunc.apply(typeWithIds);
+      this.rowGroups = reader.getRowGroups();
+      this.shouldSkip = new boolean[rowGroups.size()];
+
+      ParquetMetricsRowGroupFilter statsFilter = null;
+      ParquetDictionaryRowGroupFilter dictFilter = null;
+      if (filter != null) {
+        statsFilter = new ParquetMetricsRowGroupFilter(expectedSchema, filter, caseSensitive);
+        dictFilter = new ParquetDictionaryRowGroupFilter(expectedSchema, filter, caseSensitive);
+      }
+
+      long computedTotalValues = 0L;
+      for (int i = 0; i < shouldSkip.length; i += 1) {
+        BlockMetaData rowGroup = rowGroups.get(i);
+        boolean shouldRead = filter == null || (
+            statsFilter.shouldRead(typeWithIds, rowGroup) &&
+                dictFilter.shouldRead(typeWithIds, rowGroup, reader.getDictionaryReader(rowGroup)));
+        this.shouldSkip[i] = !shouldRead;
+        if (shouldRead) {
+          computedTotalValues += rowGroup.getRowCount();
+        }
+      }
+
+      this.totalValues = computedTotalValues;
+      this.reuseContainers = reuseContainers;
+      this.batchSize = bSize;
+    }
+
+    ReadConf(ReadConf<T> toCopy) {
+      this.reader = null;
+      this.file = toCopy.file;
+      this.options = toCopy.options;
+      this.projection = toCopy.projection;
+      this.model = toCopy.model;
+      this.rowGroups = toCopy.rowGroups;
+      this.shouldSkip = toCopy.shouldSkip;
+      this.totalValues = toCopy.totalValues;
+      this.reuseContainers = toCopy.reuseContainers;
+      this.batchSize = toCopy.batchSize;
+    }
+
+    ParquetFileReader reader() {
+      if (reader != null) {
+        reader.setRequestedSchema(projection);
+        return reader;
+      }
+
+      ParquetFileReader newReader = newReader(file, options);
+      newReader.setRequestedSchema(projection);
+      return newReader;
+    }
+
+    VectorizedReader model() {
+      return model;
+    }
+
+    boolean[] shouldSkip() {
+      return shouldSkip;
+    }
+
+    long totalValues() {
+      return totalValues;
+    }
+
+    boolean reuseContainers() {
+      return reuseContainers;
+    }
+
+    int batchSize() {
+      return batchSize;
+    }
+
+    private static ParquetFileReader newReader(InputFile file, ParquetReadOptions options) {
+      try {
+        return ParquetFileReader.open(ParquetIO.file(file), options);
+      } catch (IOException e) {
+        throw new RuntimeIOException(e, "Failed to open Parquet file: %s", file.location());
+      }
+    }
+
+    ReadConf<T> copy() {
+      return new ReadConf<>(this);
+    }
+  }
+
+  private ReadConf conf = null;
+
+  private ReadConf init() {
+    if (conf == null) {
+      ReadConf readConf = new ReadConf(
+          input, options, expectedSchema, filter, batchReaderFunc, reuseContainers, caseSensitive, batchSize);
+      this.conf = readConf.copy();
+      return readConf;
+    }
+
+    return conf;
+  }
+
+  @Override
+  public Iterator iterator() {
```

Comment:
Can you check types? I think this should be `Iterator<T>`

Reply:
Fixed